srcs/client/Client_recv.cpp

```diff
@@ -0,0 +1,282 @@
+		str_to_lower(field_name);
+		line.erase(0, i + 1);
+		_headers[field_name] = line;
+
+		if (field_name == "content-type") {
+			std::string::size_type boundary_index = line.find("boundary");
+			if (boundary_index != std::string::npos) {
+				_body_boundary = line.erase(0, boundary_index + 9);
+			}
+		}
+		return;
+	}
+	else if (line.empty()) {
+		if (_headers.count("host") == 0) // No Host received
+			throw 400;
+		_process_headers = false; // All headers has been received
+		if (_method != "POST")
+			_request_is_complete = true;
+		// Search for the configuration whose server_names match with host 
+		// by default the first server for a host:port will be
+		// the default for this host:port
+		retrieve_conf(_headers["host"]);
+
+		// Check if the method is allowed by the virtual server
+		check_method(_method);
+
+		// Check is the resource is accessible
+		check_access(_request_target);
+		return;
+	}
+	throw 400; // Malformed header
+}
+
+void Client::parse_line(std::deque<std::string> &lines, std::string &raw_request) {
+
+	remove_cr_char(lines);
+
+	while (!lines.empty()) {
+		if (_process_request_line){
+			process_request_line(lines.front());
+			lines.pop_front();
+		}
+		else if (_process_headers)
+		{
+			process_field_line(lines.front());
+			lines.pop_front();
+		}
+		else {
+			process_body(raw_request);
+			lines.clear();
+		}
+
+		if (_request_is_complete)
+			break;
+	}
+	return ;
+}
+
+std::deque<std::string> Client::getlines(std::string buf) {
+	std::deque<std::string> lines;
+	_ss << buf;
+
+	std::string::size_type i = buf.find("\n");
+	while (!buf.empty()) {
+		std::string inter;
+		std::getline(_ss, inter);
+		lines.push_back(inter);
+		buf.erase(0, i + 1);
+		i = buf.find("\n");
+		if (i == std::string::npos) {
+			i = buf.size();
+		}
+	}
+	return lines;
+}
+
+bool Client::handle_request(std::string &raw_request) {
+	// set status code too 400 when no raw_request recieved
+	if (raw_request.empty()) {
+		_request_is_complete = true;
+		_status_code = 400;
+	}
+	std::deque<std::string>	lines;
+	lines = getlines(raw_request);
+	if (!lines.empty()) {
+		try { // This will catch any bad request 
+			parse_line(lines, raw_request);
+		}
+		catch (int error_code) {
+			_status_code = error_code;
+			return true;
+		}
+	}
+
+	return _request_is_complete;
+}
+
+std::string Client::recv_request() {
+	char buffer[BUFFER_SIZE] = {};
+	int valread;
+	std::string raw_request;
+
+	while ((valread = recv(_fd, buffer, BUFFER_SIZE - 1, 0)) > 0) {
+		buffer[valread] = '\0';
+		raw_request.append(buffer, valread);
+	}
+	if (valread == 0) {
+		// change the condition from || to &&, dont know if it's logic?
+		if (_process_request_line == false && _process_headers == false)
+			_request_is_complete = true;
+	}
+	if (valread < 0) {
+		_syscall_error = "recv()";
+		log(log_error(_syscall_error), false);
+		// Still don't know why sometimes will have "Resource temporarily unavailable" error...
+		// Is something blocking the recv?
+	}
```

Comment:
I'm working on this part to remove the loop. (To avoid BLOCKING behavior?!)

Reply:
Okay, you can merge your work later, but I recommend you to use this branch